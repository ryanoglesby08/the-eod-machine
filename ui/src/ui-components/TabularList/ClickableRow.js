import React from 'react'
import PropTypes from 'prop-types'

import { Box, colors } from 'rebass/emotion'

const css = {
  font: 'inherit',
  border: 0,

  '&:hover, &:focus': {
    backgroundColor: colors.gray,
  },
}

const ClickableRow = ({ children, ...rest }) => (
  <Box {...rest} is="button" width="100%" p={3} bg="transparent" css={css}>
    {children}
  </Box>
)
ClickableRow.propTypes = {
  children: PropTypes.node.isRequired,
}

export default ClickableRow
