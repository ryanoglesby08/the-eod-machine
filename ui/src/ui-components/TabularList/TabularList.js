import React from 'react'

import styled from 'react-emotion'
import { Border } from 'rebass/emotion'

const lastItemInList = (list, index) => list.length === index + 1

const BareList = styled('ul')({
  padding: 0,
  margin: 0,
  listStyleType: 'none',
})

const TabularList = ({ rows }) => (
  <BareList>
    {rows.map((row, index) => (
      <Border
        is="li"
        py={3}
        px={2}
        borderBottom={lastItemInList(rows, index) ? undefined : 0}
        key={index}
      >
        {row}
      </Border>
    ))}
  </BareList>
)

export default TabularList
