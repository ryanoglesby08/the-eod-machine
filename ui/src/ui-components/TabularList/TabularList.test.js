import React from 'react'

import { render } from 'react-testing-library'

import TabularList from './TabularList'

it('is a bordered, unordered list that resembles a table', () => {
  const { container } = render(<TabularList rows={['a', 'b', 'c']} />)

  expect(container).toMatchSnapshot()
})
