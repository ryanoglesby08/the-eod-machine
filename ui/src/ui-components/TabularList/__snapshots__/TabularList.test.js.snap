// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is a bordered, unordered list that resembles a table 1`] = `
.emotion-3 {
  padding: 0;
  margin: 0;
  list-style-type: none;
}

.emotion-0 {
  border: 1px solid;
  border-bottom: 0;
  border-color: gray;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 16px;
  padding-bottom: 16px;
}

.emotion-2 {
  border: 1px solid;
  border-color: gray;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 16px;
  padding-bottom: 16px;
}

<div>
  <ul
    class="emotion-3"
  >
    <li
      class="emotion-0"
    >
      a
    </li>
    <li
      class="emotion-0"
    >
      b
    </li>
    <li
      class="emotion-2"
    >
      c
    </li>
  </ul>
</div>
`;
